rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,159 @@
+Here we'll have 3 types of column grids, these will be used depending on the use case.
+1. **Default:** This has a gutter of `32px` & will be used on the internal dashboard.
+![Default Column-Grid](./images/spatial/col-grid-default.png)
+
+2. **Condensed:** This has a gutter of `16px` & will be used on mobile layouts.
+![Condensed Column-Grid](./images/spatial/col-grid-condensed.png)
+
+3. **Spacious:** This has a gutter of `48px` & will be used for landing/marketing pages.
+![Condensed Column-Grid](./images/spatial/col-grid-spacious.png)
+
+These column grids can also be used internally based on the what component we are building. For example, let's say we are building inline form, then we'll use condensed col-grid for organising elements.
+
+### **Spatial Scale**
+We'll follow a **8pt** linear scale for the elements with the **4pt** half-step for spacing smaller content (icons, etc.) and typography (line-heights).
+
+This is how the scale will look like:
+
+| Alias      | Value |
+|------------|-------|
+| $spacer-0  | 0px   |
+| $spacer-1  | 4px   |
+| $spacer-2  | 8px   |
+| $spacer-3  | 12px  |
+| $spacer-4  | 16px  |
+| $spacer-5  | 20px  |
+| $spacer-6  | 24px  |
+| $spacer-7  | 32px  |
+| $spacer-8  | 40px  |
+| $spacer-9  | 48px  |
+| $spacer-10 | 56px  |
+
+>These are mixed values, it contains multiple of 4 as well as 8.
+
+# Drawbacks/Constraints
+There are no major drawbacks here,
+- We need to make sure that all the spacing between the elements should not be hardcoded. They should use the spatial tokens whenever required.
+- We might find few screens where it is difficult to organise the spacing. For example, iPhone 6 (`375x667`). For such scenarios we don't need to break the pixels, instead we need to keep the margins & paddings consistent. It's ok to have oodly-sized layout which keeps the grid consistent. Although, no user will likely see the actual measurements here.
+
+# Alternatives
+There are many other spatial systems we might have considered, such as **5pt grid**, **6pt grid**. But we are not using them because of already mentioned reason,
+- With odd numbers like 5, it is hard to align things to center as the outcome will result in fraction values that will split the pixels. This can also result in blurry pixel.
+- Similarly the scaling will also not work properly while scaling on mobile/tablets, which will again result into pixel blurriness.
+
+# Adoption strategy
+Spatial scale needs to be implemented as a core part of the DS in terms of the tokens which will automatically derive the adoption.
+- Implement the [spatial scale](#spatial-scale) (as tokens).
+- Implement the 3 types of [column-grid](#column-grid) structure using the same spatial scale.
+- Make sure all the components are using the same spatial scale tokens while building them.
+
+# How do we educate people?
+Both **designers** & **developers** need to be educated.
+- Designers should always use the spatial (multiple of 4pt) scale while working on any layout.
```

Comment:
Is there a way to enforce this on Figma?

Reply:
Hahaha! Can't enforce, but yes we can guide them through in a way where they should use the spacer grids before the implementation of the design.